runners/dmn-tck-runner/src/main/java/org/omg/dmn/tck/runner/junit4/DmnTckVendorTestSuite.java

```diff
@@ -64,10 +65,26 @@
      *                call.
      * @param testCases the parsed content of a test file.
      *
-     * @param modelURI the resolved URI to the DMN model file.
+     * @param modelURL the resolved URL to the DMN model file.
+     * @deprecated To support DMN features involving multiple models, see {@link #beforeTestCases(TestSuiteContext, TestCases, URL, List)}
      */
+    @Deprecated
     void beforeTestCases(TestSuiteContext context, TestCases testCases, URL modelURL );
 
+    /**
+     * A callback to give vendors an opportunity for initialization
+     * before running each test file.
+     *
+     * @param context the context created by the <code>createContext()</code>
+     *                call.
+     * @param testCases the parsed content of a test file.
+     *
+     * @param modelURL the resolved URL to the DMN model file.
+     */
+    default void beforeTestCases(TestSuiteContext context, TestCases testCases, URL modelURL, Collection<? extends URL> additionalModels) {
```

Comment:
In order to maintain backward compatibility, a default implementation calling back to previous method is provided. Previous method this way is supported OOTB for unchanged runners. Old runners will still see old callback, however naturally without any explicit change they won't be able to make use of the new parameter.
New runners can make use of this new method.